models/articlesModel.js

```diff
@@ -1,45 +1,71 @@
 const db = require("../db/connection.js");
 
-const fetchArticleById = (articleId) => {
-    if (isNaN(parseInt(articleId))) {
-        return Promise.reject({
-            status: 400,
-            msg: "Bad request"
-        })
-    }
-    return db.query("SELECT * FROM ARTICLES WHERE article_id = $1", [articleId])
-    .then(({rows}) => {
-        const article = rows[0]
-        if (!article) {
-            return Promise.reject({
-                status: 404,
-                msg: `No article found for article_id ${articleId}`
-            })
-        }
-        return article
-    })
-}
+const fetchArticleById = (articleId, commentCount) => {
+	if (isNaN(parseInt(articleId))) {
+		return Promise.reject({
+			status: 400,
+			msg: "Bad request",
+		});
+	}
+	if (commentCount) {
+		return db
+			.query(
+				`SELECT articles.*, COUNT(comments.article_id) AS comment_count FROM articles
```

Comment:
comment_count should be a number/int, have a look at how to cast this piece of data as a different data type